cinema/serializers.py

```
@@ -114,9 +114,9 @@ class TicketSerializer(serializers.ModelSerializer):
     def validate(self, attrs):
         data = super(TicketSerializer, self).validate(attrs=attrs)
         Ticket.validate_ticket(
-            attrs["row"], 
-            attrs["seat"], 
-            attrs["movie_session"].cinema_hall, 
+            attrs.get("row"),
+            attrs.get("seat"),
+            attrs.get("movie_session").cinema_hall,
             ValidationError
         )
```

Error: In the `validate` method, you are calling `attrs.get("row")`, `attrs.get("seat")`, and `attrs.get("movie_session").cinema_hall` without checking if these fields are present. If any of them are missing, this will cause a `KeyError` or an `AttributeError`. You should check for their presence and raise a `ValidationError` if any are missing, as required by robust serializer validation.